mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -2690,49 +2747,65 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
-        } else if (arg2TargetLayoutName2 == "gemmK") {
-          isOutputPad = true;
-          isGemmDim2Pad = true;
-          gemmDim2TargetName = b.getStringAttr(gemmKPad_name);
-
-          paddingOutputShape[2] = paddingOutputShape[2] + gemmKExtra;
-          sourceGemmDim2Attr.push_back(
-              b.getNamedAttr("transformation", b.getStringAttr("Pad")));
-          sourceGemmDim2Attr.push_back(b.getNamedAttr(
-              "parameters", b.getArrayAttr({b.getI32IntegerAttr(0),
-                                            b.getI32IntegerAttr(gemmKExtra)})));
-
-          targetGemmDim2Attr.push_back(
-              b.getNamedAttr("upper_layer_names",
-                             b.getArrayAttr({b.getStringAttr(gemmKPad_name)})));
+          // wrw
+          outputOobCheckDims.insert(nameToDims["no"]);
+          outputOobCheckDims.insert(nameToDims["ho"]);
+          outputOobCheckDims.insert(nameToDims["wo"]);
         }
-        // output of forward, gemmK = no * ho * wo
-        outputOobCheckDims.insert(nameToDims["no"]);
-        outputOobCheckDims.insert(nameToDims["ho"]);
-        outputOobCheckDims.insert(nameToDims["wo"]);
       }
 
-      if (gemmMExtra > 0) {
+      if (outputCheckPadGemmM) {
         if (arg2TargetLayoutName1 == "gemmM") {
-          isOutputPad = false;
-          isGemmDim1Pad = false;
+          isOutputPad = true;
+          isGemmDim1Pad = true;
+          gemmDim1TargetName = b.getStringAttr(gemmMPad_name);
+          // fwd, k
           paddingOutputShape[1] = paddingOutputShape[1] + gemmMExtra;
+          sourceGemmDim1Attr.push_back(
+              b.getNamedAttr("transformation", b.getStringAttr("Pad")));
+          sourceGemmDim1Attr.push_back(b.getNamedAttr(
+              "parameters", b.getArrayAttr({b.getI32IntegerAttr(0),
+                                            b.getI32IntegerAttr(gemmMExtra)})));
+
+          targetGemmDim1Attr.push_back(b.getNamedAttr(
+              "names", b.getArrayAttr({b.getStringAttr(gemmMPad_name)})));
+          outputOobCheckDims.insert(nameToDims["ko"]);
         } else if (arg2TargetLayoutName2 == "gemmM") {
-          isOutputPad = false;
-          isGemmDim2Pad = false;
+          isOutputPad = true;
+          isGemmDim2Pad = true;
+          gemmDim2TargetName = b.getStringAttr(gemmMPad_name);
+          // wrw ,k
           paddingOutputShape[2] = paddingOutputShape[2] + gemmMExtra;
+          sourceGemmDim2Attr.push_back(
+              b.getNamedAttr("transformation", b.getStringAttr("Pad")));
+          sourceGemmDim2Attr.push_back(b.getNamedAttr(
+              "parameters", b.getArrayAttr({b.getI32IntegerAttr(0),
+                                            b.getI32IntegerAttr(gemmMExtra)})));
+
+          targetGemmDim2Attr.push_back(b.getNamedAttr(
+              "names", b.getArrayAttr({b.getStringAttr(gemmMPad_name)})));
+          outputOobCheckDims.insert(nameToDims["ko"]);
```

Comment:
please add comments here on the rationale assigning dimensions to OOB check dimensions, like what you did above for filter.

Reply:
comment was not added.